packages/jest-fake-timers/src/jestFakeTimers.ts

```diff
@@ -177,11 +173,8 @@ export default class FakeTimers<TimerRef> {
   }
 
   private _runImmediate(immediate: Tick) {
-    if (!this._cancelledImmediates.hasOwnProperty(immediate.uuid)) {
-      // Callback may throw, so update the map prior calling.
-      this._cancelledImmediates[immediate.uuid] = true;
-      immediate.callback();
-    }
+    immediate.callback();
```

Comment:
`try-finally`? Ref the deleted comment

Reply:
swapped lines

Reply:
Looks like it's the wrong way around again and no try finally either? That's probably why e2e fails